python/packages/kagent-adk/src/kagent/adk/_agent_executor.py

```diff
@@ -256,11 +256,24 @@ async def _prepare_session(self, context: RequestContext, run_args: dict[str, An
             session_id=session_id,
         )
         if session is None:
+            # Extract session name from the first TextPart (like the UI does)
+            session_name = None
+            if context.message and context.message.parts:
+                for part in context.message.parts:
+                    # A2A parts have a .root property that contains the actual part (TextPart, FilePart, etc.)
+                    if isinstance(part, Part):
+                        root_part = part.root
+                        if isinstance(root_part, TextPart) and root_part.text:
+                            # Take first 20 chars + "..." if longer (matching UI behavior)
+                            text = root_part.text.strip()
+                            session_name = text[:20] + ("..." if len(text) > 20 else "")
+                            break
             session = await runner.session_service.create_session(
                 app_name=runner.app_name,
                 user_id=user_id,
                 state={},
                 session_id=session_id,
+                session_name=session_name,
```

Comment:
nit: this is calling the base class which doesn't have a `session_name` attribute, so this code may fail if running with a different `session_service`. I hate Python, but can we use the `*` argument from the args somehow?